Docs/LibraryArchitecture.md

```diff
@@ -74,13 +93,13 @@ Requirement <|-- AccessTokenRequirement: implements
 <<Interface>> Requirement
 class Requirement {
     +required: Bool
-    +validate()
+    +validate() Result<<T>Void>
```

Comment:
nit: At some future point we should probably include the error type so the intention of this function is clear.